From akpm@linux-foundation.org Mon Mar 17 14:21:46 2008
From: Boaz Harrosh <bharrosh@panasas.com>
Date: Mon, 17 Mar 2008 14:21:01 -0700
Subject: USB: isd200: fix memory leak in isd200_get_inquiry_data
To: greg@kroah.com
Cc: akpm@linux-foundation.org, bharrosh@panasas.com, James.Bottomley@HansenPartnership.com, stern@rowland.harvard.edu
Message-ID: <200803172121.m2HLL2Ks031648@imap1.linux-foundation.org>


From: Boaz Harrosh <bharrosh@panasas.com>

If the inquiry fails then the info structure on us->extra was not freed.

Signed-off-by: Boaz Harrosh <bharrosh@panasas.com>
Acked-by: Alan Stern <stern@rowland.harvard.edu>
Cc: James Bottomley <James.Bottomley@HansenPartnership.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/isd200.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/storage/isd200.c
+++ b/drivers/usb/storage/isd200.c
@@ -1230,6 +1230,7 @@ static int isd200_get_inquiry_data( stru
 	    
 			/* Free driver structure */	    
 			us->extra_destructor(info);
+			kfree(info);
 			us->extra = NULL;
 			us->extra_destructor = NULL;
 		}
